qpylib/live_qpylib.py

```diff
@@ -9,102 +9,101 @@
-APP_CERT_LOCATION = '/etc/pki/tls/certs/ca-bundle.crt'
-
 class LiveQpylib(AbstractQpylib):
+    LOGFILE_LOCATION = '/store/log/app.log'
+    APP_CERT_LOCATION = '/etc/pki/tls/certs/ca-bundle.crt'
+    APP_MANIFEST_LOCATION = 'app/manifest.json'
+
+    QRADAR_CONSOLE_FQDN = 'QRADAR_CONSOLE_FQDN'
+    QRADAR_CSRF = 'QRadarCSRF'
+    SEC_HEADER = 'SEC'
+    SEC_ADMIN_TOKEN = 'SEC_ADMIN_TOKEN'
+
+    APP_FILE_LOG_FORMAT = '%(asctime)s [%(module)s.%(funcName)s] [%(threadName)s] [%(levelname)s] - %(message)s'
+    APP_CONSOLE_LOG_FORMAT = '%(asctime)s %(module)s.%(funcName)s: %(message)s'
+    
+    # ==== Logging ====
+
+    def _add_log_handler(self, loc_logger):
+        loc_logger.setLevel(self._map_log_level(self._get_manifest_field_value('log_level', 'info')))
+        
+        handler = RotatingFileHandler(self.LOGFILE_LOCATION, maxBytes=2*1024*1024, backupCount=5)
+        handler.setFormatter(Formatter(self.APP_FILE_LOG_FORMAT))
+        loc_logger.addHandler(handler)
 
-    def get_manifest_location(self):
-        return 'app/manifest.json'
-
-    def get_console_address(self):
-        manifest = self.get_manifest_json()
-        console_ip = ''
-        if 'console_ip' not in manifest.keys():
-            self.log('console not defined in manifest - default to localhost', level='error')
-            console_ip = '127.0.0.1'
-        else:
-            console_ip = manifest['console_ip']
-        return console_ip
-
-    def get_cert_filepath(self, host=None):
-        # host is sdk-specific
-        if '/etc/qradar_pki' in open('/proc/mounts').read():
-            self.log('Using ca bundle cert from file: ' + str(APP_CERT_LOCATION), level='debug')
-            cert_filepath = APP_CERT_LOCATION
-        else:
-            self.log('/etc/qradar_pki is not mounted in the container, validation turned off', level='debug')
-            cert_filepath = False
-        return cert_filepath
-
-    def acquire_SEC_token(self):
-        return request.cookies.get('SEC')
+        # Ipv6 address - Strip [] for syslog
+        console_ip = self.get_console_address()
+        if console_ip.startswith('[') and console_ip.endswith(']'):
+            console_ip = console_ip[1:-1]
 
-    def acquire_QRadarCSRF_token(self):
-        return request.cookies.get('QRadarCSRF')
+        syslogHandler = SysLogHandler(address=(console_ip, 514))
+        syslogHandler.setFormatter(Formatter(self.APP_CONSOLE_LOG_FORMAT))
+        loc_logger.addHandler(syslogHandler)
+        return
 
-    def get_tokens(self, headers, version=None):
-        csrf_cookie = 'QRadarCSRF'
-        sec_cookie = 'SEC'
+    # ==== App details ====
 
-        if headers is None:
-            headers = {}
+    def get_app_id(self):
+        return self._get_manifest_field_value('app_id', 0)
 
-        if version is not None:
-            headers['Version'] = version
+    def get_app_name(self):
+        return self._get_manifest_field_value('name')
+    
+    def _get_manifest_location(self):
```

Comment:
Not sure if this is necessary as it was done for the SDK

Reply:
I'll check and remove it if not necessary.